addons/udes_report/models/stock_export.py

```diff
@@ -165,35 +199,47 @@ def run_movement_file_export(self, context=None, send_file_via='user'):
         in_moves = moves.filtered(lambda m: m.picking_type_id == picking_type_in)
         out_moves = moves.filtered(lambda m: m.picking_type_id == picking_type_out)
 
-        _logger.info(_('Creating Excel file'))
-        file_name = 'warehouse_movement_{TIMESTAMP}.xls'.format(
-            TIMESTAMP=datetime.now())
-        wb = xlwt.Workbook()
-
-        for goods_moves, sheet_name in [(in_moves, "Goods In"),
-                                        (out_moves, "Goods Out")]:
-            sheet = _create_sheet(wb, sheet_name, ['Reference',
-                                                   'Part number',
-                                                   'Package',
-                                                   'Quantity'])
-
-            row = 0
+        data = []
+        for goods_moves, sheet_name in [(in_moves, "Goods In"), (out_moves, "Goods Out")]:
+            move_data = {
+                "name": sheet_name,
+                "column_titles": ['Reference', 'Part number', 'Package', 'Quantity']
+            }
+            rows = []
             for move in goods_moves:
                 for move_line in move.mapped('move_line_ids'):
-                    row += 1
-                    sheet.write(row, 0, move.picking_id.origin)
-                    sheet.write(row, 1, move.product_id.display_name)
-                    sheet.write(row, 2, move_line.result_package_id.name)
-                    sheet.write(row, 3, move_line.qty_done)
+                    rows.append(
+                        {'Reference': move.picking_id.origin,
+                         'Part number': move.product_id.display_name,
+                         'Package': move_line.result_package_id.name,
+                         'Quantity': move_line.qty_done}
+                    )
+            move_data['rows'] = rows
+            data.append(move_data)
 
-        self._write_workbook(wb, file_name, "Movement File",
-                             send_file_via=send_file_via)
+        return data
 
     @api.model
-    def _write_workbook(self, workbook, file_name, doc_title,
-                        send_file_via='user'):
+    def _write_workbook(self, file_name, doc_title, data, send_file_via="user"):
```

Comment:
Data is an existing variable that is used later (line 197). I would suggest to rename one of them.

Reply:
Renamed